Various fixes

tofu-rocketry opened the pull request:
Resolves GT-1240
Resolves #63 (by speeding up the queries to an acceptable level)

This PR does the following:
- Replace some of the queries with much faster ones.
- Fixes some of the logic around which result is fetched for the benchmarks page so that you actually get the most relevant row for each site and submithost.
- Removes redundant `print()` that were used when debugging during initial creation.
- Fixes the comparisons for the status evaluation to allow future dates and add a critical level.
- Tweaks the readme.
- Also applies a small SQL change to the IRIS update script to try and improve its performance a bit.

 ## Speed comparison
### Before
Over 1 hour.
```
2025-11-27 07:01:02,050 - INFO - =====================
2025-11-27 07:01:11,655 - INFO - Refreshed GridSite
2025-11-27 07:01:31,630 - INFO - Refreshed GridSiteSync
2025-11-27 07:03:01,444 - INFO - Refreshed GridSiteSyncSubmitH
2025-11-27 07:17:13,359 - INFO - Refreshed CloudSite
2025-11-27 07:17:41,885 - INFO - Refreshed BenchmarksBySubmitHost from VSummaries
2025-11-27 08:17:42,443 - INFO - Refreshed BenchmarksBySubmitHost from VJobRecords
2025-11-27 08:17:51,911 - INFO - Refreshed BenchmarksBySubmitHost from VNormalisedSummaries
2025-11-27 08:17:51,911 - INFO - Data retrieval and processing attempted. Check the above logs for details on the sync status
2025-11-27 08:17:51,911 - INFO - =====================
```

### After
Less than 5 minutes.
```
2025-12-12 20:47:16,860 - INFO - =====================
2025-12-12 20:47:19,483 - INFO - Refreshed GridSite
2025-12-12 20:47:32,894 - INFO - Refreshed GridSiteSync
2025-12-12 20:48:29,054 - INFO - Refreshed GridSiteSyncSubmitH
2025-12-12 20:49:06,731 - INFO - Refreshed CloudSite
2025-12-12 20:49:07,980 - INFO - Refreshed BenchmarksBySubmitHost from VSummaries
2025-12-12 20:51:31,971 - INFO - Refreshed BenchmarksBySubmitHost from VJobRecords
2025-12-12 20:51:32,210 - INFO - Refreshed BenchmarksBySubmitHost from VNormalisedSummaries
2025-12-12 20:51:32,210 - INFO - Data retrieval and processing attempted. Check the above logs for details on the sync status
2025-12-12 20:51:32,210 - INFO - =====================
```

tofu-rocketry committed:
Update readme

tofu-rocketry committed:
Replace cloud update SQL with more efficient query

- This version uses windowing (SELECT...ROW_NUMBER() OVER) and a smaller
  range of UpdateTime for efficiency.
- It uses a Common Table Expression (WITH...AS) for  readability.
- The first row for each site is then selected to give the latest values
  for that site.
- Counting the VMs is deprecated as it's not efficient to calculate and
  doesn't add much for the user. Removing it here should default the
  value to 0 in the model.

tofu-rocketry committed:
Remove VMs column from cloud page

Previous commit removes the update of this data. We're deprecating it as
it doesn't add anything useful for the user at the moment and was
affecting the SQL performance.

tofu-rocketry committed:
Replace JobRecords benchmark SQL with better query

- This version uses windowing (SELECT...ROW_NUMBER() OVER) and a smaller
  range of UpdateTime for efficiency.
- It uses a Common Table Expression (WITH...AS) for readability.
- The first row for each site and submit host is then selected to give
  the latest values for that site/submithost combination.

tofu-rocketry committed:
Improve benchmark SQL for summary tables

These changes are similar to teh Job Records imporvements, but
customised for the way these views work.

tofu-rocketry committed:
Update to right no. of months on benchamark page

tofu-rocketry committed:
Remove conditional prints left from debugging

These were used during initial creation of the system to debug update
times, but aren't needed in production, particularly as we use a
completely different method to update the local Django database.

tofu-rocketry committed:
Tweak warning function

- Allow for future dates to be considered OK.
- Add a critical threshold.

tofu-rocketry committed:
Flip status logic so that it actually works

tofu-rocketry commented:
<strike>TODO: Should improve the IRIS VAnonCloudRecords query</strike> - Done

tofu-rocketry committed:
Correct periods stated in page templates

tofu-rocketry committed:
Apply a moderate performance fix to IRIS script

rowan04 commented on monitoring/db_update_sqlite.py:
```diff
@@ -176,31 +176,30 @@ def refresh_gridsite():
                 FROM VAnonCloudRecords
-                WHERE UpdateTime > DATE_SUB(NOW(), INTERVAL 1 YEAR)
-                GROUP BY SiteName
+                WHERE UpdateTime >= CURRENT_TIMESTAMP - INTERVAL 6 MONTH
```
Why is `CURRENT_TIMESTAMP` used here, instead of `NOW()` which is used elsewhere?

tofu-rocketry replied:
Yeah, I should just make it consistent. One of them is an older alias, but I need to look up which is the preferred one these days!

tofu-rocketry replied:
Done in [c75aadb](https://github.com/apel/monitoring/pull/73/commits/c75aadb461d17465f7230e748af1ef09a3d693eb)

rowan04 commented on monitoring/db_update_sqlite.py:
```diff
@@ -305,53 +304,82 @@ def refresh_BenchmarksBySubmitHost_from_view(view_name):
+                        UpdateTime,
+                        ROW_NUMBER() OVER (
+                            PARTITION BY Site, SubmitHost
+                            ORDER BY Year DESC, Month DESC, UpdateTime DESC
```
```suggestion
                            ORDER BY UpdateTime DESC
```
why is the `Year DESC` and `Month DESC` needed? Will it not just sort out into the same order?

tofu-rocketry replied:
We want to make sure we're looking at the data (particularly the benchmark info) most recently published for the most recent month, not just the most recently published for any month.

rowan04 commented on monitoring/db_update_sqlite.py:
```diff
@@ -305,53 +304,82 @@ def refresh_BenchmarksBySubmitHost_from_view(view_name):
+                        UpdateTime,
+                        ROW_NUMBER() OVER (
+                            PARTITION BY Site, SubmitHost
+                            ORDER BY Year DESC, Month DESC, UpdateTime DESC
```
```suggestion
                            ORDER BY UpdateTime DESC
```
same as above

tofu-rocketry replied:
As above. 😛

rowan04 commented on monitoring/db_update_sqlite.py:
```diff
@@ -305,53 +304,82 @@ def refresh_BenchmarksBySubmitHost_from_view(view_name):
+                            ORDER BY Year DESC, Month DESC, UpdateTime DESC
+                        ) AS rn
+                    FROM VNormalisedSummaries
+                    WHERE UpdateTime >= NOW() - INTERVAL 2 MONTH
```
swapping `DATE_SUB()` for this looks good - would it be beneficial to do this on line 159 as well (I think 159 is the only place `DATE_SUB` is used in this file now)?

tofu-rocketry replied:
Done in [c75aadb](https://github.com/apel/monitoring/pull/73/commits/c75aadb461d17465f7230e748af1ef09a3d693eb)

rowan04 reviewed (commented):
looks good, some very small comments

tofu-rocketry committed:
Standardise time comparisons

- Change last remaining CURRENT_TIMESTAMP to NOW().
- Replace DATE_SUB with plain subtraction of timestamp.

rowan04 reviewed (approved):
lgtm :)
